jaeger-core/src/test/java/io/jaegertracing/JaegerSpanTest.java

```diff
@@ -444,6 +509,38 @@ public void testNoExpandExceptionLogs() {
     assertEquals(ex, logData.get(0).getFields().get(Fields.ERROR_OBJECT));
   }
 
+  @Test
+  public void testNoLoggingingStackTracesOnDebug() {
+    log.setEnabledLevels(Level.WARN);
+    JaegerSpan jaegerSpan = (JaegerSpan)tracer.buildSpan("foo").start();
+    jaegerSpan.finish();
+    jaegerSpan.finish();
+
+    assertEquals(1, log.getLoggingEvents().size());
+    LoggingEvent event = log.getLoggingEvents().get(0);
+    assertEquals("Span has already been finished; cannot {}.", event.getMessage());
+    assertEquals(Level.WARN, event.getLevel());
+    assertEquals(false, event.getThrowable().isPresent());
+  }
+
+  @Test
+  public void testLoggingingStackTracesOnDebug() {
+    log.setEnabledLevels(Level.WARN, Level.DEBUG);
+    JaegerSpan jaegerSpan = (JaegerSpan)tracer.buildSpan("foo").start();
```

Comment:
Same as the previous comment.

Reply:
There have to be a few more changes for this to be possible (buildSpan has to return a `JaegerTracer.SpanBuilder`, and `SpanBuilder` needs to be public.

I'll make these changes, but I don't know if you want to do that in a separate diff

Reply:
No no, just rebase to the latest master. Those changes are already there :)

https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/io/jaegertracing/JaegerTracer.java#L401